common/djangoapps/student/views.py

```diff
@@ -2315,15 +2315,32 @@ def activate_account(request, key):
     else:
         if not registration.user.is_active:
             registration.activate()
-            # Add account activation success message for display later
-            messages.success(
-                request,
-                HTML(_('{html_start}Success{html_end} You have activated your account.')).format(
-                    html_start=HTML('<p class="message-title">'),
-                    html_end=HTML('</p>'),
-                ),
-                extra_tags='account-activation aa-icon',
-            )
+
+            if request.user.is_authenticated():
+                # Add account activation success message for logged in users
+                messages.success(
+                    request,
+                    HTML(_('{html_start}Success{html_end} You have activated your account.')).format(
+                        html_start=HTML('<p class="message-title">'),
+                        html_end=HTML('</p>'),
+                    ),
+                    extra_tags='account-activation aa-icon',
+                )
+            else:
+                # Add account activation success message for logged out users
+                messages.success(
+                    request,
+                    HTML(_(
+                        '{html_start}Success! You have activated your account.{html_end}'
+                        'You will now receive email updates and alerts from us related to'
+                        ' the courses you are enrolled in. Sign In to continue.'
+                    )).format(
+                        html_start=HTML('<p class="message-title">'),
+                        html_end=HTML('</p>'),
+                    ),
+                    extra_tags='account-activation aa-icon',
+                )
```

Comment:
Can you just set a variable containing the message string inside the if/else? Then we only need to write the larger code block once.